src/kernel/tcap.c

```diff
@@ -215,6 +218,30 @@ tcap_merge(struct tcap *dst, struct tcap *rm)
 	return 0;
 }
 
+int
+tcap_wakeup(struct tcap *tc, tcap_prio_t prio, struct thread *thd, struct cos_cpu_local_info *cli)
+{
```

Comment:
Updated PR. I'll try and add a test case for the budget test in `cos_thd_wakeup()`.